vaadin-combo-box-flow/src/test/java/com/vaadin/flow/component/combobox/ComboBoxTest.java

```diff
@@ -283,11 +283,12 @@ public void setValueNull_selectedItemNull() {
     }
 
     @Test
-    public void setValueWithoutItems_throw() {
-        expectIllegalStateException(
-                "Cannot set a value for a ComboBox without items.");
+    public void setValueWithoutItems_valueIsSet() {
         ComboBox<String> combo = new ComboBox<>();
         combo.setValue("foo");
+        // Since the combo box is always initialised with data communicator on
+        // instantiation phase, setting the value is possible
+        Assert.assertEquals("foo", combo.getValue());
```

Comment:
How can a combo box without any items have a selected item as its value ??? What does that mean ? When would it be useful instead of just a developers error ? Do we support that for other selection components too ?

It feels to me that you accidentally enabled this by internal changes and it should not still be possible

Reply:
The related feature for this is allowing new items and I don't think this ^ is the way to do it

Reply:
I thought that since the combo box now has a `dataCommunicator` instance being created in the CTOR, we could remove  `if (dataCommunicator == null)` code block. And the old `setValue()` allowed new values on data communicator with no items, but not null. However, you're right, there is no sense to set value to empty combo box. So, I propose: 1. Revert the removed code block, since `setValue` still can be called before DC init accidentally  2. Add the items count check.

Reply:
Done